### PR TITLE
Add test for column level constraint for LIKE for AI collations

### PR DESCRIPTION
**Pull request by ahmed-shameem**
### Description
This commit adds JDBC tests for column level constrains using CHECK while inserting value into a table. If the CHECK condition is not met, user won't be able to insert data into table
Upgrade test for 15_latest to 16_latest has also been added.

### Issues Resolved

BABEL-4931
Signed-off-by: Shameem Ahmed [shmeeh@amazon.com](mailto:shmeeh@amazon.com)

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/collation.c [hunk 1]
```diff
@@ -410,19 +410,22 @@ transform_from_ci_as_for_likenode(Node *node, OpExpr *op, like_ilike_info_t like
...
-											InvalidOid, server_collation_oid, oprfuncid(optup));
+											InvalidOid, coll_info_of_inputcollid.oid, oprfuncid(optup));
 		ReleaseSysCache(optup);
 		/* construct pattern||E'\uFFFF' */
-		highest_sort_key = makeConst(TEXTOID, -1, server_collation_oid, -1,
+		highest_sort_key = makeConst(TEXTOID, -1, coll_info_of_inputcollid.oid, -1,
 										PointerGetDatum(cstring_to_text(SORT_KEY_STR)), false, false);
+		
+		/* Set the collation explicitly for the Const node */
+		highest_sort_key->constcollid = coll_info_of_inputcollid.oid;
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/collation.c:420]: why is this needed?

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/collation.c (hunk 1 above)
Comment (ahmed-shameem) [reply contrib/babelfishpg_tsql/src/collation.c:420]: This is redundant, I was testing something on my part. Will remove it in next revision.
